src/realtime/updater.py

```
@@ -424,6 +526,119 @@ def process_parsed_records_batch(self, records: list[Dict], timeseries: bool = F
+
+            if owns_transaction:
+                self.database.commit()
+            total_errors = validation_errors + prepared_errors
+            return {
+                "operation": "batch_insert",
+                "success": total_errors == 0,
+                "inserted": inserted,
+                "errors": total_errors,
+                "tables": sorted(tables),
+            }
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)